library/src/main/java/io/github/boguszpawlowski/composecalendar/week/WeekListState.kt

```diff
@@ -0,0 +1,74 @@
+package io.github.boguszpawlowski.composecalendar.week
+
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.snapshotFlow
+import io.github.boguszpawlowski.composecalendar.states.CurrentState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+@Stable
+internal class WeekListState(
```

Comment:
This and `WeekPager` could be easily used in order to create separate `WeekCalendar` composable.

Reply:
Why do it separately when you can combine it?